src/sparseml/pytorch/torchvision/train.py

```diff
@@ -418,7 +424,25 @@ def collate_fn(batch):
             model_without_ddp, device=device, decay=1.0 - alpha
         )
 
-    if args.resume:
+    manager = ScheduledModifierManager.from_yaml(args.recipe_path)
+
+    if args.checkpoint_path:
+        checkpoint = torch.load(args.checkpoint_path, map_location="cpu")
+        model_without_ddp.load_state_dict(checkpoint["model"])
+        if not args.test_only:
+            optimizer.load_state_dict(checkpoint["optimizer"])
+            lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+        args.start_epoch = checkpoint["epoch"] + 1
+        if model_ema:
+            model_ema.load_state_dict(checkpoint["model_ema"])
+        if scaler:
+            scaler.load_state_dict(checkpoint["scaler"])
+
+        checkpoint_manager = ScheduledModifierManager.from_yaml(
+            checkpoint["checkpoint_recipe"]
+        )
+        checkpoint_manager.apply_structure(model_without_ddp, epoch=checkpoint["epoch"])
```

Comment:
all model loading should go through here for consistency: https://github.com/neuralmagic/sparseml/blob/21b48cc4fa8e6a1ba183b6f1ad0fb221b9fc3ff8/src/sparseml/pytorch/utils/model.py#L58

Reply:
bumping this since marked outdated